### PR TITLE
roles: ad, adding ADSites class to create new sites objects

### PR DESCRIPTION
**Pull request by danlavu**

### CODE REVIEW

**Comment by danlavu** [comment]
GPO tests need to be able to create a new site. Also the restore, cn=sites,cn=configuration is in the config part of the directory and needs to be explicitly set to be searched.

**Review by pbrezina — DISMISSED** [review]

File: sssd_test_framework/roles/ad.py [hunk 1]
```diff
@@ -281,6 +281,33 @@ def test_example(client: Client, ad: AD):
         """
         return ADOrganizationalUnit(self, name, basedn)
 
+    def site(self, name: str, basedn: ADObject | str | None = "cn=sites,cn=configuration") -> ADSite:
+        """
+        Get site object.
+
+        .. code-blocK:: python
```
Comment (spoore1) [inline sssd_test_framework/roles/ad.py:288]: Change to lowercase K in code-blocK?

**Review by spoore1 — CHANGES_REQUESTED** [review]
Just one very minor nitpick for a docstring.

**Review by spoore1 — APPROVED** [review]
LGTM

**Pull request merged**